### ISSUE TITLE
Improve atomicity of slashing protection DB creation

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#1537]
Several users on Discord have reported their VC being unable to start because of missing tables in the slashing protection SQLite database.

As far as I can tell, this can only happen if the VC is stopped in the first moments of startup, after it has created the SQLite DB but before it has created the tables.

One way to fix it might be to create the DB in a temporary location and then use an atomic filesystem operation to move it into position

**Comment by paulhauner** [comment]
Hey @michaelsproul do you think we need this before the Oct sec review?

**Comment by michaelsproul** [comment]
@paulhauner I don't think we need it per-se, it's at most a minor annoyance and not a security risk
